src/Quereum.sol

```diff
@@ -137,14 +125,17 @@ contract Quereum {
             status: 0, // status is active
             user: msg.sender, // the user posting the question
             reward: reward, // the reward for the question
-            responses: new uint256, // initializing responses array
+            responses: new uint256[](0), // initializing responses array
             rewardAllocated: false // reward not allocated yet
         });
 
         // add the question to the array of questions and the user's questions
         questions.push(newQuestion);
         userQuestions[msg.sender].push(questions.length - 1);
 
+        // deduct the reward from the user's balance
+        balances[msg.sender] -= reward;

```

Comment:
I'm not sure but shouldn't we withdraw the amount from the sender before we create the object that way they cant self destruct the operation before it removes the value?

Reply:
changed!